ContactEmailHandlerFactory.cls

```diff
@@ -0,0 +1,31 @@
+/**
+* @description: This is the data factory for Contact_Email_Handler_Junction__c 
+                object where dummy records are created and inserted and furthur 
+                used in the test class.
+* @author: Noopur Naik
+*
+* @revision:
+* Version      Date            Description
+* ------------------------------------------------
+* 1.0          30-Jun-2017     Initial revision.
+**/
+@isTest
+public class ContactEmailHandlerFactory {
```

Comment:
Why there is another `TestDataFactory`. In org there should be only one data factory and which can be use by all the test classes.